### PR TITLE
Convert Spacer component to TypeScript

### LINKED ISSUE #181: Convert Spacer component to TypeScript

### PR DESCRIPTION
**Pull request by ivanleeswe**
Convert Spacer component to TypeScript

### CODE REVIEW

File: src/components/grid-aware/Spacer/Spacer.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  heightMobile,
+  color = "transparent",
+}: SpacerProps) => (
+  <div
+    style={
+      {
+        "--spacer-height-desktop": heightDesktop,
+        "--spacer-height-mobile": heightMobile,
+        "--spacer-color": color,
+      } as React.CSSProperties
```
Comment (richardxia) [inline src/components/grid-aware/Spacer/Spacer.tsx:22]: Ah, this is interesting. I hadn't thought about how the CSS custom properties would interact with TypeScript. I think this approach is OK for CSS custom properties that are local to just that component, but if we ever find ourselves using custom properties across multiple components, then we may want to try out an approach closer to https://stackoverflow.com/a/65959390, where they defined a custom interface that extends the `CSSProperties` interface.

**Review by richardxia — APPROVED** [review]
LGTM!

File: src/components/grid-aware/Spacer/Spacer.tsx (hunk 1 above)
Comment (davidagustin) [reply src/components/grid-aware/Spacer/Spacer.tsx:22]: Interesting, forgot about this mechanic also.

**Review by davidagustin — APPROVED** [review]

**Review by davidagustin — APPROVED** [review]

**Review by jessica-px — APPROVED** [review]

**Review by candywang — APPROVED** [review]
LGTM!

**Pull request merged**